midend/interpreter.cpp

```diff
@@ -309,22 +309,38 @@ void SymbolicStruct::dbprint(std::ostream& out) const {
     out << " }";
 }
 
-SymbolicHeaderUnion::SymbolicHeaderUnion(const IR::Type_HeaderUnion* type,
-                               bool uninitialized,
-                               const SymbolicValueFactory* factory) :
-        SymbolicStruct(type, uninitialized, factory),
-        valid(new SymbolicBool(false)) {}
+SymbolicHeaderUnion::SymbolicHeaderUnion(const IR::Type_HeaderUnion *type,
+                                         bool uninitialized,
+                                         const SymbolicValueFactory *factory)
+    : SymbolicStruct(type, uninitialized, factory),
+      valid(new SymbolicBool(false)) {
+  auto fieldsClone = fieldValue;
+  bool validityFlag = false;
+  for (auto f : type->to<IR::Type_StructLike>()->fields) {
+    if (fieldsClone[f->name.name]->to<SymbolicHeader>()->valid->value) {
+      validityFlag = true;
+      break;
+    }
+  }
+  valid = new SymbolicBool(validityFlag);
+}
 
-void SymbolicHeaderUnion::setValid(bool v) {
-    if (!v)
-        setAllUnknown();
-    valid = new SymbolicBool(v);
+void SymbolicHeaderUnion::setValid(bool v, cstring field) {
+  if (field) {
+    fieldValue[field]->to<SymbolicHeader>()->setValid(v);
+    for (auto f : type->to<IR::Type_StructLike>()->fields) {
+      if (f->name.name != field)
+        fieldValue[f->name.name]->setAllUnknown();
+    }
+  }
+  valid = new SymbolicBool(v);
 }
 
 SymbolicValue* SymbolicHeaderUnion::get(const IR::Node* node, cstring field) const {
-    if (valid->isKnown() && !valid->value)
-        return new SymbolicStaticError(node, "Reading field from invalid header union");
-    return SymbolicStruct::get(node, field);
+  if (valid->isKnown() && !valid->value)
```

Comment:
I don't really understand you. How hard is it to use 4 spaces for indentation?
So let's say that you do use 2 spaces only, but why reindent code that you don't even modify?
Can you please check the PRs before you submit them to see if indentation is wrong?

Reply:
I checked with cpplint and clang-format and they didn't show any indentation errors, but if there are I'll correct them manually

Reply:
I made indentation fixes